src/rating-indicator.scss

```diff
@@ -0,0 +1,190 @@
+@import "./new-settings";
+@import "./mixins";
+@import "./icon";
+
+$block: #{$fd-namespace}-rating-indicator;
+
+.#{$block} {
+  $unratedColor: var(--sapContent_UnratedColor);
+  $fd-rating-indicator-sizes: (
+    "cozy": ("fontSize": 1.5rem, "spacingBetweenIcons": 0.1875rem, "controlMargin": 0.8125rem 0),
+    "compact": ("fontSize": 1rem, "spacingBetweenIcons": 0.125rem, "controlMargin": 0.5rem 0),
+    "condensed": ("fontSize": 1rem, "spacingBetweenIcons": 0.125rem),
+    "xs": ("fontSize": 0.75rem, "spacingBetweenIcons": 1px),
+    "s": ("fontSize": 1rem, "spacingBetweenIcons": 0.125rem),
+    "m": ("fontSize": 1.375rem, "spacingBetweenIcons": 0.1875rem),
+    "l": ("fontSize": 2rem, "spacingBetweenIcons": 0.25rem)
+  );
+  $fd-rating-indicator-default-size: map-get($fd-rating-indicator-sizes, 'm');
+  $fd-rating-indicator-default-font-size: map_get($fd-rating-indicator-default-size, "fontSize");
+  $fd-rating-indicator-default-spacing-between-icons: map_get($fd-rating-indicator-default-size, "spacingBetweenIcons");
+
+  @mixin fd-rating-indicator-with-half-star($iconFontSize, $spacingBetweenIcons) {
+    .#{$block}__input {
+      width: $iconFontSize / 2;
+      height: $iconFontSize;
+    }
+
+    .#{$block}__label {
+      margin-right: 0;
+
+      &:nth-child(4n+4) {
+        margin-right: $spacingBetweenIcons;
+
+        &::before {
+          float: right;
+        }
+      }
+
+      overflow: hidden;
+      width: $iconFontSize / 2;
+    }
+  }
+
+  @include fd-reset();
+
+  position: relative;
+
+  &__container {
+    @include fd-fiori-focus-within();
```

Comment:
I know `focus-within` is great and it fits well for this case, but it's not supported by IE11. You can achieve same thing by having pseudo element on label, when input is focused. So `.fd-input:focus + .fd-label::after { position: absolute ...;  } with position relative on container element

Reply:
extended documentation
thank you for the comments 🙂